api/views/default.py

```diff
@@ -240,6 +240,7 @@ def class_detail_list(request):
                 "deadline": assignment.deadline,
                 "max_points": assignment.max_points,
                 "students": {s["username"]: {"username": s["username"]} for s in students},
+                "task_type": "" if assignment.task.type is None else assignment.task.type,
```

Comment:
`""` should be converted to JSON `null` and handled the normal way in JavaScript, right? I don't think we need to introduce another special empty value. Or was there some special reason why use the empty string?